corehq/apps/locations/models.py

```diff
@@ -660,9 +636,30 @@ def full_delete(self):
         with transaction.atomic():
             for loc in to_delete:
                 loc._close_case_and_remove_users()
-            SQLLocation.objects.get(location_id=self._id).delete()
+            # delete all SQLLocations without calling their `delete` methods
+            self.sql_location.get_descendants(include_self=True).delete()
             Location.get_db().bulk_delete(to_delete)
```

Comment:
I think I might make this the default delete behavior (since `SQLLocation.delete` cascades anyways), but that feels a bit outside the scope of this PR